dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+        Parameter covariance for each frequency and input.
+    """
+    # Check if boolean flag was input
+    if flag is None:
+        flag = np.ones(response.shape, dtype=np.bool)
+    elif flag.dtype != np.bool:
+        flag = flag.astype(np.bool)
+
+    # Create arrays to hold best-fit parameters and
+    # parameter covariance.  Initialize to NaN.
+    nfreq = response.shape[0]
+    npol = response.shape[1]
+    nparam = 6  # magic number!!
+
+    parameter = np.full((nfreq, npol, nparam), np.nan, dtype=np.float32)
+    parameter_cov = np.full((nfreq, npol, nparam, nparam), np.nan, dtype=np.float32)
+    resid_rms = np.full((nfreq, npol), np.nan, dtype=np.float32)
+
+    # Create initial guess at FWHM if one was not input
+    if fwhm is None:
+        fwhm = np.full((nfreq, npol), 3.0, dtype=np.float32)
+
+    # Iterate over frequency/pol and fit point source transit
+    for ff in range(nfreq):
+
+        for pp in range(npol):
+
+            this_flag = flag[ff, pp]
+
+            # Only perform fit if there is enough data.
+            # Otherwise, leave parameter values as NaN.
+            if np.sum(this_flag) <= nparam:
+                continue
+
+            # We will fit the complex data.  Break n-element complex array g(ra)
+            # into 2n-element real array [Re{g(ra)}, Im{g(ra)}] for fit.
+            x = hour_angle[this_flag]
+
+            y = response[ff, pp, this_flag]
+            yerr = np.sqrt(tools.invert_no_zero(weight[ff, pp, this_flag]))
+
+            # Initial estimate of parameter values
+            p0 = np.array(
+                [
+                    np.max(y) - np.median(y),
+                    np.median(x),
+                    fwhm[ff, pp],
+                    np.asarray([np.median(y), 0.0, 0.0]),
+                ]
+            )
+
+            # Perform the fit.  If there is an error,
+            # then we leave parameter values as NaN.
+            try:
+                popt, pcov = curve_fit(
+                    _fit_func1, x, y, p0=p0, sigma=yerr, absolute_sigma=False
+                )
+            except Exception as error:
+                continue
+
+            # Save the results
+            parameter[ff, pp] = popt
+            parameter_cov[ff, pp] = pcov
+
+            # Compute scatter in residuals
+            resid = y - _fit_func1(x, *popt)
+
+            resid_rms[ff, pp] = np.std(resid_dict)
+
+    # Return the best-fit parameters and parameter covariance
+    return parameter, parameter_cov, resid_rms
+
+
+def _fit_func1(x, peak_amplitude, p0):
+
+    centroid, fwhm, bpoly = p0
+    dx = x - centroid
+    return _func_gauss(dx, peak_amplitude, fwhm) + _func_background(dx, bpoly)
+
+
+def _func_gauss(dx, peak_amplitude, fwhm):
+
+    return peak_amplitude * np.exp(-4.0 * np.log(2.0) * (dx / fwhm) ** 2)
+
+
+def _func_background(dx, bck):
+
+    model = np.polyval(bck, dx)
+    return model
+
+
+def _correct_phase_wrap(phi, deg=False):
```

Comment:
You might consider making this a helper function.  I think `feedpositions_analyzer` is using an equivalent on line 364.